# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/chromium/hotfix-rust-t2.patch
# Copyright (C) 2026 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- chromium-151.0.7922.75/build/config/rust.gni.vanilla
+++ chromium-151.0.7922.75/build/config/rust.gni
@@ -251,6 +251,10 @@
     # Best guess for other future platforms.
     rust_abi_target = current_cpu + "-unknown-linux-gnu"
   }
+
+  if (current_cpu == target_cpu && t2_rust_abi_target != "") {
+    rust_abi_target = t2_rust_abi_target
+  }
 } else if (is_android) {
   import("//build/config/android/abi.gni")
   if (android_abi_target == "i686-linux-android") {
--- chromium-151.0.7922.75/build/rust/std/BUILD.gn.vanilla
+++ chromium-151.0.7922.75/build/rust/std/BUILD.gn
@@ -74,7 +74,6 @@
   # don't need to pass to the C++ linker because they're used for specialized
   # purposes.
   skip_stdlib_files = [
-    "profiler_builtins",
     "rustc_std_workspace_alloc",
     "rustc_std_workspace_core",
     "rustc_std_workspace_std",
